clinica/utils/filemanip.py

```diff
@@ -1,209 +1,476 @@
+        id_bids_or_caps_files.append(m.group())
+
     return id_bids_or_caps_files
 
 
-def extract_subjects_sessions_from_filename(bids_or_caps_files):
-    """Extract subjects/sessions (e.g. ['sub-CLNC01', 'sub-CLNC01']/['ses-M00', 'ses-M18'] from `bids_or_caps_files`."""
+def extract_subjects_sessions_from_filename(
+    bids_or_caps_files: List[str],
+) -> Tuple[List[str], List[str]]:
+    """Extract the subject and session labels from a list of BIDS or CAPS files.
+
+    Parameters
+    ----------
+    bids_or_caps_files: List[str]
+        List of file paths for which to extract the subject and session labels.
+
+    Returns
+    -------
+    subject_ids: List[str]
+        List of subject labels in the same order as `bids_or_caps_files`.
+
+    session_ids: List[str]
+        List of session labels in the same order as `bids_or_caps_files`.
+
+    Examples
+    --------
+    >>> extract_subjects_sessions_from_filename([
+    ...     "sub-01/ses-M000/pet/sub-01_ses-M000_trc-18FAV45_pet.nii.gz",
+    ...     "foo/bar/baz/sub-foo/ses-bar/foooo/sub-01_ses-M000_foo.json", "sub-01_ses-M000.tar.gz",
+    ...])
+    (['sub-01', 'sub-01', 'sub-01'], ['ses-M000', 'ses-M000', 'ses-M000'])
+
+    See also
+    --------
+    extract_image_ids
+    """
     id_bids_or_caps_files = extract_image_ids(bids_or_caps_files)
     split = [image_id.split("_") for image_id in id_bids_or_caps_files]
     subject_ids = [p_id[0] for p_id in split]
     session_ids = [s_id[1] for s_id in split]
     return subject_ids, session_ids
 
 
-def extract_crash_files_from_log_file(filename):
-    """Extract crash files (*.pklz) from `filename`."""
-    import os
-    import re
+def extract_crash_files_from_log_file(filename: str) -> List[str]:
+    """Extract crash files (*.pklz) from `filename`.
 
-    assert os.path.isfile(
-        filename
-    ), f"extract_crash_files_from_log_file: filename parameter is not a file ({filename})"
+    Parameters
+    ----------
+    filename: str
+        Path to log file for which the crash files should be extracted.
 
-    log_file = open(filename, "r")
+    Returns
+    -------
+    crash_files: List[str]
+        List of crash files.
+    """
+    import re
+    from pathlib import Path
+
+    filename = Path(filename)
+    if not filename.is_file():
+        raise ValueError(
+            f"extract_crash_files_from_log_file: filename parameter is not a file ({filename})"
+        )
     crash_files = []
-    for line in log_file:
-        if re.match("(.*)crashfile:(.*)", line):
-            crash_files.append(line.replace("\t crashfile:", "").replace("\n", ""))
+    with open(filename, "r") as log_file:
+        for line in log_file:
+            if re.match("(.*)crashfile:(.*)", line):
+                crash_files.append(line.replace("\t crashfile:", "").replace("\n", ""))
 
     return crash_files
 
 
-def read_participant_tsv(tsv_file):
+def read_participant_tsv(tsv_file: str) -> Tuple[List[str], List[str]]:
```

Comment:
Just a side note, we could ensure both lists are of the same length by returning a `Iterable[Tuple[str, str]]` instead of a tuple of iterables.

Reply:
Do you suggest we do this modification in this PR ?
I think it is a good idea but I expect a few changes on code relying on this function. I'm OK for doing it here or in a follow-up PR, as you prefer.

Reply:
Yes, we would need to trace where `read_participant_tsv` is used and where its output are tested. I agree that this should be the scope of another PR.